modules/compiler/utils/include/compiler/utils/replace_wgc_pass.h

```diff
@@ -39,7 +39,13 @@ namespace utils {
 /// target making use of that pass.
 class ReplaceWGCPass final : public llvm::PassInfoMixin<ReplaceWGCPass> {
  public:
+  ReplaceWGCPass(bool scansOnly = false) : scans_only(scansOnly) {}
   llvm::PreservedAnalyses run(llvm::Module &, llvm::ModuleAnalysisManager &);
+
+ private:
+  /// @brief when set to true, only work group collective scans will be
+  /// replaced.
+  bool scans_only = false;
```

Comment:
What about a property in `BIMuxInfoImpl` that requests full software implementations of the WGC functions, which RefSi G1 WIPT sets and the `ReplaceWGCPass` reads? That way we don't need to anticipate the barriers pass running.

Or a bitfield of which ones the target would like to be provided for it: `unsigned please_give_me = reductions | broadcasts | scans`

That way it's not so much as pass pipeline thing as it is a target configuration.

Reply:
i just wanted to do something quickly that worked because (a) i'm about to go away and (b) this whole thing is supposed to be got rid of before too long anyway. it's only hanging around at all because i didn't have time to get Scans working the same way.

Reply:
if you want to do this then by all means, it could be useful for other things who knows, but getting the scans into the barrier pass would be more of a priority and then we won't need to parameterize the replace-wgc pass at all, we would simply not run it when using the barrier pass, in fact it should by that point be possible to run it after vecz rather than before like now so in effect it just becomes a replacement for/alternative to the barrier pass.

Resolving comment now.